### ISSUE TITLE
SearchForm should have `role="search"`

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#6680]
The `SearchForm` component should have a `role="search"` so that a11y tools know that the form is designed for search purpose.

**Comment by ioanarusiczki** [comment]
@willdurand Should we check on regressions for the search component?

**Comment by willdurand** [comment]
> @willdurand Should we check on regressions for the search component?

Yes, but a quick check is fine

**Comment by ioanarusiczki** [comment]
No regressions found. Verified on AMO dev with FF62 (Win10)
